sqlglot/parser.py

```diff
@@ -3394,8 +3394,9 @@ def _parse_cte(self) -> t.Optional[exp.CTE]:
             comments=comments,
         )
 
-        if isinstance(cte.this, exp.Values):
-            cte.set("this", exp.select("*").from_(exp.alias_(cte.this, "_values", table=True)))
+        values = cte.this
+        if isinstance(values, exp.Values) and not values.alias:
+            cte.set("this", exp.select("*").from_(exp.alias_(values, "_values", table=True)))
```

Comment:
Why not reuse the `TableAlias` attached to the `Values` node instead? This is going to generate inconsistent SQL depending on the presence (or not) of an alias, right? Doesn't seem right.